File: app/src/main/java/com/gals/prayertimes/Splashscreen.java
package com.gals.prayertimes;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;


public class Splashscreen extends AppCompatActivity {
    public static final String PREFS_NAME = "MyLocalStorage";
    Prayer prayer;
    SharedPreferences settings;
    ToolsManager tools;
    Intent toMain;
    //TODO: change to database.
    //TODO: handel exceptions well.
    //TODO: Testing.
    //TODO: remove old code.
    //TODO: background services.



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(com.gals.prayertimes.R.layout.activity_splashscreen);

        toMain = new Intent(this, MainActivity.class);
        settings = getSharedPreferences(PREFS_NAME, 0);
        prayer = new Prayer(settings, this.getBaseContext());
        tools = new ToolsManager(getBaseContext());

        // Change Statusbar color
        tools.changeStatusBarColor(getWindow());

        GetServerData task = new GetServerData(toMain, prayer, settings, this.getBaseContext(), false);
        task.execute("", "", "");

        /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            prayer.setPrayersAlarms();
            Log.i("ALARM", "From Splash Screen");
        }*/

    }
}

File: app/src/main/java/com/gals/prayertimes/view/Splashscreen.kt
package com.gals.prayertimes.view

 import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.gals.prayertimes.R
import com.gals.prayertimes.utils.UtilsManager
import com.gals.prayertimes.viewmodel.SplashscreenViewModel
import com.gals.prayertimes.viewmodel.factory.SplashscreenViewModelFactory
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@SuppressLint("CustomSplashScreen")
@AndroidEntryPoint
class Splashscreen : AppCompatActivity() {
    private lateinit var toMain: Intent
    private lateinit var viewModel: SplashscreenViewModel
    lateinit var tools: UtilsManager

    @Inject
    lateinit var viewModelFactory: SplashscreenViewModelFactory

    //TODO: handle exceptions well and handle no network situation.
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splashscreen)

        tools = UtilsManager(baseContext)

        configureMVVM()
        configureLoading()

        /**Change Status bar color*/
        tools.changeStatusBarColor(window)

        viewModel.prepareApp()
    }

    private fun configureMVVM() {
        viewModel = ViewModelProvider(
            this,
            viewModelFactory
        )[SplashscreenViewModel::class.java]
    }

    private fun configureLoading() {
        viewModel.loading.observe(this, Observer { status ->
            if (!status) {
                navigateToMain()
            }
        }
        )
    }

    private fun navigateToMain() {
        toMain = Intent(
            this,
            MainActivity::class.java
        )
        toMain.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
        this.startActivity(toMain)
    }
}